runtimes/dart-2.18/function_types.dart

```diff
@@ -1,33 +1,77 @@
 import 'dart:convert';
 
 class Request {
-  final Map<String, dynamic> variables;
-  final Map<String, dynamic> headers;
-  final String payload;
-
-  Request({
-    this.variables = const {},
-    this.headers = const {},
-    this.payload = '',
-  });
+  String bodyString;
+  dynamic body;
+  Map<String, dynamic> headers;
+  String method;
+  String url;
+  String path;
+  int port;
+  String scheme;
+  String host;
+  String queryString;
+  Map<String, String> query;
```

Comment:
Why don't we use `URI` instead of this? `URI` is not supported in any other language?

Reply:
We decided to not use URI component and stay consistent across all runtimes.